src/cargo/ops/cargo_new.rs

```diff
@@ -795,8 +795,27 @@ fn mk(config: &Config, opts: &MkOptions<'_>) -> CargoResult<()> {
                     .push(bin);
             }
         } else if i.relative_path != "src/lib.rs" {
-            manifest["lib"]["name"] = toml_edit::value(i.target_name.clone());
-            manifest["lib"]["path"] = toml_edit::value(i.relative_path.clone());
+            let mut lib = toml_edit::Table::new();
+            lib["name"] = toml_edit::value(i.target_name.clone());
+            lib["path"] = toml_edit::value(i.relative_path.clone());
+            manifest["lib"] = toml_edit::Item::Table(lib);
```

Comment:
Similarly, this looks unrelated to the current commit

Reply:
Fixed.